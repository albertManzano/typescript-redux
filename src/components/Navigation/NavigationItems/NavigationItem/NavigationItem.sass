.NavigationItem
    margin: 10px 0
    box-sizing: border-box
    display: block
    width: 100%
    align-items: center
    &__link
        color: #8f5c2c
        text-decoration: none
        width:  100%
        box-sizing: border-box
        display: block
        &--active
            color: #40a4c8
        &:hover, &:active
            color: #40a4c8

@media (min-width:500px)
    .NavigationItem
        margin: 0
        box-sizing: border-box
        display: flex
        width: auto
        height: 100%
        align-items: center
        &__link
            color: white
            text-decoration: none
            height: 100%
            width: auto
            padding: 16px 10px
            box-sizing: border-box
            display: block
            border-bottom: 4px solid transparent
            &--active
                background-color: #8f5c2c
                border-bottom: 4px solid #40a4c8
                color: white
                padding: 16px 10px
            &:hover, &:active
                background-color: #8f5c2c
                border-bottom: 4px solid #40a4c8
                color: white
