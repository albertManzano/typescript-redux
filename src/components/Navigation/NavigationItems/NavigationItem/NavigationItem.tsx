import React from 'react'/* eslint-disable-line */

import'./NavigationItem.sass';

const NavigationItem = ({link,children,active}) =>  <li className='NavigationItem'><a className={active ? 'NavigationItem__link--active' : 'NavigationItem__link'} href={link}>{children}</a></li>;

export default NavigationItem;