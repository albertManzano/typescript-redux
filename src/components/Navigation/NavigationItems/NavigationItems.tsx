import React from 'react';/* eslint-disable-line */
import './NavigationItems.sass'


import NavigationItem from './NavigationItem/NavigationItem'

const NavigationItems = () =>{
    return (
        <ul className='NavigationItems'>
          <NavigationItem link='/' active={false} >Burguer Constructor</NavigationItem>
          <NavigationItem link='/orders' active={false} >Orders</NavigationItem>
        </ul>
    )

};

export default NavigationItems;