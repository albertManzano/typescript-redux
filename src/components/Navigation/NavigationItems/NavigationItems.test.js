import React from 'react'; /* eslint-disable-line */

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import NavigationItems from './NavigationItems';
import NavigationItem from './NavigationItem/NavigationItem';

configure({ adapter: new Adapter() });

describe('<NavigationItems />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<NavigationItems />);
  });

  it('should exist', () => {
    expect(wrapper.html()).toBeDefined();
  });
  it('should render two <NavigationItem /> ', () => {
    expect(wrapper.find(NavigationItem)).toHaveLength(2);
  });
  it('should render always', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
