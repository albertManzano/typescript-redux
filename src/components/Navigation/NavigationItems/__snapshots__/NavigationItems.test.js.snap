// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationItems /> should render always 1`] = `ShallowWrapper {}`;
